### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by fmterrorf**
This adds the initial code for the Java agent. 

**Testing**

You need to install Maven and Java11

There is an `example` project where all it does is make an http request. To use the agent with it, you have to do the following

1. Package the `example` app to a jar by doing `(cd example && mvn clean package)`
2. Package the gent and run it with the example jar

```
mvn clean package &&
  java -javaagent:target/metrist-java-agent-0.0.1.jar \
  -jar example/target/example_agent_usage-1.0-SNAPSHOT.jar
```

This should send a UDP packet to the orchestrator

### CODE REVIEW

File: src/main/java/io/metrist/agent/MetristAgent.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    HttpRequest request = response.request();
+    URL url = request.uri().toURL();
+    String message = new StringBuilder("0\t")
+        .append(request.method())
+        .append("\t")
+        .append(url.getHost())
+        .append("\t")
+        .append(url.getPath())
+        .append("\t")
+        .append(System.nanoTime() - startTime)
```
Comment (cdegroot) [inline src/main/java/io/metrist/agent/MetristAgent.java]: Does this resolve to (fractional) milliseconds?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots package
+      - run: mkdir staging && cp target/*.jar staging
+
+      - name: Release
```
Comment (cdegroot) [inline .github/workflows/main.yml]: Let's remove this for now, not useful to spread artifacts all over the place without an (automated) install process and no signing. We have dist.metrist.io if we want to distribute stuff outside the stack's regular channels.

File: src/main/java/io/metrist/agent/MetristAgent.java (hunk 1 above)
Comment (fmterrorf) [reply src/main/java/io/metrist/agent/MetristAgent.java]: I have a commit i forgot to push that changes this to millis https://github.com/Metrist-Software/java-agent/pull/1/commits/b2cbf958bf227898c9f8c9e6317bccb98cae7502 :facepalm:

File: .github/workflows/main.yml (hunk 2 above)
Comment (fmterrorf) [reply .github/workflows/main.yml]: Removed and moved it to dist.metrist.io

**Review by cdegroot — APPROVED** [review]

**Pull request merged**